hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -501,6 +501,23 @@ private ContainerCommandResponseProto runCommand(
     return raftFuture;
   }
 
+  private StateMachine.DataChannel getStreamDataChannel(
+          ContainerCommandRequestProto requestProto,
+          DispatcherContext context) {
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("{}: getStreamDataChannel {} containerID={} pipelineID={} " +
+                      "traceID={}", gid, requestProto.getCmdType(),
+              requestProto.getContainerID(), requestProto.getPipelineID(),
+              requestProto.getTraceID());
+    }
+    try {
+      runCommand(requestProto, context);  // stream init
+      return dispatcher.getStreamDataChannel(requestProto);
+    } catch (IOException e) {
+      throw new CompletionException("Failed to create data stream", e);
+    }
```

Comment:
Do not try-catch and throws StorageContainerException in the getStreamDataChannel(..) declaration.